### PR TITLE
sys: Fix warning for stripped const qualifier

### PR DESCRIPTION
**Pull request by yperess**
Fix several errors caused by adding stricter warnings used by pigweed

Signed-off-by: Yuval Peress <peress@google.com>

### CODE REVIEW

File: include/sys/cbprintf_internal.h [hunk 1]
```diff
@@ -314,7 +314,7 @@ union z_cbprintf_hdr {
  * if argument is a stirng literal. Suppression is added here instead of
  * the macro which generates the warning to not slow down the compiler.
  */
-#if __clang__ == 1
+#if defined(__clang__) && __clang__ == 1
```
Comment (jackrosenthal) [inline include/sys/cbprintf_internal.h:317]: `#ifdef __clang__`

File: include/logging/log_msg2.h [hunk 2]
```diff
@@ -28,7 +28,7 @@ extern "C" {
 #define LOG_MSG2_DEBUG 0
 #define LOG_MSG2_DBG(...) IF_ENABLED(LOG_MSG2_DEBUG, (printk(__VA_ARGS__)))
 
-#if CONFIG_LOG_TIMESTAMP_64BIT
+#ifdef CONFIG_LOG_TIMESTAMP_64BIT
```
Comment (jackrosenthal) [inline include/logging/log_msg2.h:31]: these unrelated changes seem fine, but are they necessary for the missing const?

File: include/sys/cbprintf_cxx.h [hunk 3]
```diff
@@ -52,6 +52,7 @@ static inline int z_cbprintf_cxx_is_pchar(const volatile wchar_t *)
 template < typename T >
 static inline int z_cbprintf_cxx_is_pchar(T arg)
 {
+	ARG_UNUSED(arg);
```
Comment (jackrosenthal) [inline include/sys/cbprintf_cxx.h:55]: same as this, seems unrelated to the missing const (should be separate PR)?

**Review by jackrosenthal — CHANGES_REQUESTED** [review]

File: include/sys/cbprintf_cxx.h (hunk 3 above)
Comment (nordic-krch) [reply include/sys/cbprintf_cxx.h:55]: I think that it's fine as long as it's in separate commit.

File: include/logging/log_msg2.h (hunk 2 above)
Comment (nordic-krch) [reply include/logging/log_msg2.h:31]: yes, you could split those into separate commit.

File: include/sys/cbprintf_cxx.h (hunk 3 above)
Comment (yperess) [reply include/sys/cbprintf_cxx.h:55]: This is in its own commit (https://github.com/zephyrproject-rtos/zephyr/pull/39932/commits/018b440cc9e950c61634578036f46e6358227a6f)

File: include/logging/log_msg2.h (hunk 2 above)
Comment (yperess) [reply include/logging/log_msg2.h:31]: Done

File: include/sys/cbprintf_internal.h (hunk 1 above)
Comment (yperess) [reply include/sys/cbprintf_internal.h:317]: Done

**Review by jackrosenthal — APPROVED** [review]

**Review by MaureenHelm — COMMENTED** [review]
Need to fix the failed compliance check:
```
Error: See docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details

Commit d67b18498a:
1: UC6 Body has no content, should at least have 1 line.

Commit 40f5517568:
1: UC6 Body has no content, should at least have 1 line.
```

**Review by ElectricalPaul — APPROVED** [review]

**Review by ElectricalPaul — APPROVED** [review]

**Review by ElectricalPaul — APPROVED** [review]

**Review by ElectricalPaul — APPROVED** [review]

**Review by ElectricalPaul — APPROVED** [review]

**Review by ElectricalPaul — APPROVED** [review]

**Comment by ElectricalPaul** [comment]
Sorry about the approval spam. I reviewed each individual commit in the stack and approved them one at a time. Apparently, approving applies to the entire stack, not to an individual commit. I guess I need to learn Github's system a little better ...

**Comment by yperess** [comment]
> Need to fix the failed compliance check:
> 
> ```
> Error: See docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details
> 
> Commit d67b18498a:
> 1: UC6 Body has no content, should at least have 1 line.
> 
> Commit 40f5517568:
> 1: UC6 Body has no content, should at least have 1 line.
> ```

Done

File: include/sys/cbprintf_cxx.h [hunk 4]
```diff
@@ -57,6 +58,7 @@ static inline int z_cbprintf_cxx_is_pchar(const volatile wchar_t *, bool const_a
 template < typename T >
 static inline int z_cbprintf_cxx_is_pchar(T arg, bool const_as_fixed)
 {
+	ARG_UNUSED(arg);
 	_Pragma("GCC diagnostic push")
```
Comment (nordic-krch) [inline include/sys/cbprintf_cxx.h:62]: can you put ARG_UNUSEDs next to each other? You can move `ARG_UNUSED(const_as_fixed);` up as well.

**Review by nordic-krch — APPROVED** [review]
looks ok, just one minor comment

**Review by dcpleung — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
